pkg/platform/runtime/setup/setup.go

```diff
@@ -160,6 +165,13 @@ func NewWithModel(target Targeter, msgHandler Events, model ModelProvider, an an
 
 // Update installs the runtime locally (or updates it if it's already partially installed)
 func (s *Setup) Update() error {
+	if s.target.OnlyInstallFromDir() != nil {
+		return s.installFromDir()
+	}
+	return s.updateFromRecipe()
```

Comment:
I feel like we were saying we want to avoid this type of logic. Basically I don't want any logic of the kind "if this is an offline install then do this, otherwise do that".

Instead sourcing of online/offline materials should be hidden behind a new interface which can be satisfied by an online and an offline implementation. Basically any sort of offline and online type logic should be centralized on this layer.

I though this is where we landed after we discussed it last time on zoom. Happy to discuss more if you don't agree with this approach or want more details.